app-backend/export/src/main/scala/com/azavea/rf/export/Export.scala

```diff
@@ -0,0 +1,146 @@
+        val (bucket, prefix) = S3.parse(exportLayerDef.ingestLocation)
+        val reader = S3LayerReader(bucket, prefix)
+        (reader, reader.attributeStore)
+      case "file" =>
+        val reader = FileLayerReader(exportLayerDef.ingestLocation.getPath)
+        (reader, reader.attributeStore)
+    }
+
+  def exportProject(params: CommandLine.Params)(exportDefinition: ExportDefinition)(implicit @transient sc: SparkContext) = {
+    val wrappedConfiguration = HadoopConfiguration(S3.setCredentials(sc.hadoopConfiguration))
+    val (input, output) = exportDefinition.input -> exportDefinition.output
+    val rdds = input.layers.map { ld =>
+      val (reader, attributeStore) = getRfLayerManagement(ld)
+      val layerId = LayerId(ld.layerId.toString, input.resolution)
+      val md = attributeStore.readMetadata[TileLayerMetadata[SpatialKey]](layerId)
+      val hist = ld.colorCorrections.map { _ => attributeStore.read[Array[Histogram[Double]]](LayerId(layerId.name, 0), "histogram") }
+      val crs = output.crs.getOrElse(md.crs)
+
+      val query = {
+        val q = reader.query[SpatialKey, MultibandTile, TileLayerMetadata[SpatialKey]](layerId)
+        input.mask.fold(q)(mp => q.where(Intersects(mp.reproject(LatLng, md.crs))))
+      } result
+
+      val bandsQuery = query.withContext {
+        _.mapValues { tile =>
+          val ctile = (ld.colorCorrections |@| hist) map { _.colorCorrect(tile, _) } getOrElse tile
+
+          output.render.flatMap(_.bands.map(_.toSeq)) match {
+            case Some(seq) if seq.nonEmpty => ctile.subsetBands(seq)
+            case _ => ctile
+          }
+        }
+      }
+
+      output.rasterSize match {
+        case Some(rs) =>
+          bandsQuery.reproject(ZoomedLayoutScheme(crs, rs))._2
+        case _ =>
+          bandsQuery.reproject(ZoomedLayoutScheme(crs))._2
+      }
+    }
+
+    /** Tile merge with respect to layer initial ordering */
+    if(!output.stitch) {
+      val result =
+        rdds
+          .zipWithIndex
+          .map { case (rdd, i) =>
+            val md = rdd.metadata
+            rdd.map { case (key, tile) => (key, i -> GeoTiff(tile, md.mapTransform(key), md.crs)) }
+          }
+          .reduce(_ union _)
+          .combineByKey(createTiles[(Int, MultibandGeoTiff)], mergeTiles1[(Int, MultibandGeoTiff)], mergeTiles2[(Int, MultibandGeoTiff)])
+          .mapValues { seq =>
+            val sorted = seq.sortBy(_._1).map(_._2)
+            val head = sorted.head
+            GeoTiff(sorted.map(_.tile).reduce(_ merge _), head.extent, head.crs)
+          }
+
+      result.foreachPartition { iter =>
+        val configuration = wrappedConfiguration.get
+        iter.foreach { case (key, tile) =>
+          tile.write(
+            new Path(s"${output.source.toString}/${input.resolution}-${key.col}-${key.row}.tiff"),
+            configuration
+          )
+        }
+      }
+    } else {
+      val md = rdds.map(_.metadata).reduce(_ combine _)
+      val tile =
```

Comment:
real nitpick, but `tile` is a little bit misleading, its already a `Raster` here.